pkg/proctree/tree/tree.go

```diff
@@ -0,0 +1,190 @@
+package tree
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strconv"
+	"sync"
+
+	"github.com/olekukonko/tablewriter"
+
+	"github.com/aquasecurity/tracee/pkg/proctree/nodes"
+)
+
+type Tree struct {
+	// TODO processes map will be a LRU map with eviction by cache limit and time
+	processes map[uint32]*nodes.Process // map[murmur3 hash of process] -> process node
```

Comment:
So currently the only way to query a process is by its hash. In signatures i'm not sure we will have access to that, and anyway a more user friendly way to consume the data source would be by a pid (maybe a composite key but i'm not sure the timestamp can be really used in such a context easily).

So my question is, for the data source usecase, how will the process tree support alternative key queries?

Reply:
@AlonZivony @roikol your input is important for this one, perhaps only the hash would suffice.

Reply:
I thought about this and thought about keeping a hash map if latest pids and their hashes. It could even have all hashes that the pid ever had by time. 

Also please note that the time series will still exist and be kept inside the Process node like the original idea. Still a TODO

Reply:
Aaaah ok I see. I was actually wondering where the TimeSeries code went. Good to know you have it in mind.

Reply:
@AlonZivony See here for the query by pid (I guess tid would also be needed).

Reply:
Yep, my intent was actually provide to each event: "task and task starttime", "parent and parent starttime". This allows you to calculate the hash of a task and its parent and no need for an extra map (or synchronize 2 LRUs caches, or have GC). This way we would have only the hashes. In the beginning of the decode event, I would calculate "task hash" "parent hash" and you would find the task and its parent. 

But then, I would have to check with you Alon: if you could provide "questions" you will pick from the datasource. Example:
- I want PID XXX.
- I want the COMM of the PARENT of this task.
- I want all opened files by this task (or by its parent ?).
I mean, a set of questions you could "ask" in the signature and that should be satisfied by the data structure/methods.